Code/Framework/AzFramework/AzFramework/Quality/QualitySystemComponent.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzFramework/Quality/QualitySystemBus.h>
+
+namespace AZ
+{
+    class IConsole;
+    class SettingsRegistryInterface;
+}
+
+namespace AzFramework
+{
+    class QualityCVarGroup;
+
+    // QualitySystemComponent manages quality groups, levels and cvar settings
+    // stored in the SettingsRegistry
+    class QualitySystemComponent final
+        : public AZ::Component
+        , public AzFramework::QualitySystemEvents::Bus::Handler
+    {
+    public:
+        AZ_COMPONENT(QualitySystemComponent, "{CA269E6A-A420-4B68-93E9-2E09A604D29A}", AZ::Component);
+
+        QualitySystemComponent();
+        ~QualitySystemComponent() override;
+
+        AZ_DISABLE_COPY(QualitySystemComponent);
+
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+        static void GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent);
+        static void Reflect(AZ::ReflectContext* context);
+
+        // AZ::Component
+        void Activate() override;
+        void Deactivate() override;
+
+        // AzFramework::QualitySystemBus
+        void LoadDefaultQualityGroup(int qualityLevel = QualitySystemEvents::LevelFromDeviceRules) override;

```

Comment:
You can make the quality level an enum as well.
Ala how [std::byte](https://en.cppreference.com/w/cpp/types/byte) is a enum class backed with operator overloads

For example you can use the AZ_TYPE_SAFE_INTEGRAL macro in [TypeSafeIntegral.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/RTTI/TypeSafeIntegral.h) to make an integer type that has no enum values. 
Or you can manually add an enum class that also supports the arithmetic and relational operators along with SerializeContext reflection which would allow users to use the enum string value for the selecting levels in the settings file itself.
i.e `"Default": "High"`

```c++
AZ_ENUM_CLASS(QualityLevels,
    (LevelFromDeviceRules, -1)
);

AZ_DEFINE_ENUM_ARITHMETIC_OPERATORS(QualityLevels);
AZ_DEFINE_ENUM_RELATIONAL_OPERATORS(QualityLevels);
AZ_ENUM_DEFINE_REFLECT_UTILITIES(QualityLevels);
```

In the `QualitySystemComponent::Reflect` function, you can invoke `QualityLevelsReflect(*serializeContext);` to reflect the enum type to the SerializeContext.

With that, you can read the the Quality level value from either a string or an integer transparently by using the SettingsRegistry::GetObject function in this case.
```c++
QualityLevels qualityLevel{ QualityLevels::LevelFromDeviceRules};
settingsRegistry->GetObject(qualityLevel, "<key>");
```

Reply:
That sounds really cool! Will give that a try.